From db5aee9c777b59676758c1225ba8de8e4408e07d Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 14:58:10 -0400
Subject: [infiniband] usnic: Update ABI and Version file for UDP support

Message-id: <e8bb3ef562ff8c38f2ac8e0f9b70bd6d6776b375.1411569661.git.dledford@redhat.com>
Patchwork-id: 96579
O-Subject: [Patch RHEL7 08/29] IB/usnic: Update ABI and Version file for UDP support
Bugzilla: 916382
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Upinder Malhi <umalhi@cisco.com>

Expand the kernel/userspace interface so userspace may push down
a socket file descriptor to usNIC.  Also, bump up the abi and version
numbers.

Signed-off-by: Upinder Malhi <umalhi@cisco.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit b85caf479b577f000067002259539ad4341d4530)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/usnic/usnic.h b/drivers/infiniband/hw/usnic/usnic.h
index d741c76..5be13d8 100644
--- a/drivers/infiniband/hw/usnic/usnic.h
+++ b/drivers/infiniband/hw/usnic/usnic.h
@@ -23,7 +23,7 @@
 
 #define PCI_DEVICE_ID_CISCO_VIC_USPACE_NIC 0x00cf /* User space NIC */
 
-#define DRV_VERSION    "1.0.2"
-#define DRV_RELDATE    "September 09, 2013"
+#define DRV_VERSION    "1.0.3"
+#define DRV_RELDATE    "December 19, 2013"
 
 #endif /* USNIC_H_ */
diff --git a/drivers/infiniband/hw/usnic/usnic_abi.h b/drivers/infiniband/hw/usnic/usnic_abi.h
index dd3e4c6..014bcfc 100644
--- a/drivers/infiniband/hw/usnic/usnic_abi.h
+++ b/drivers/infiniband/hw/usnic/usnic_abi.h
@@ -21,7 +21,7 @@
 #define USNIC_ABI_H
 
 /* ABI between userspace and kernel */
-#define USNIC_UVERBS_ABI_VERSION 2
+#define USNIC_UVERBS_ABI_VERSION 3
 
 #define USNIC_QP_GRP_MAX_WQS  8
 #define USNIC_QP_GRP_MAX_RQS  8
@@ -30,7 +30,8 @@
 enum usnic_transport_type {
  USNIC_TRANSPORT_UNKNOWN  = 0,
  USNIC_TRANSPORT_ROCE_CUSTOM = 1,
- USNIC_TRANSPORT_MAX  = 2,
+ USNIC_TRANSPORT_IPV4_UDP = 2,
+ USNIC_TRANSPORT_MAX  = 3,
 };
 
 struct usnic_transport_spec {
@@ -39,9 +40,16 @@ struct usnic_transport_spec {
   struct {
    uint16_t port_num;
   } usnic_roce;
+  struct {
+   uint32_t sock_fd;
+  } udp;
  };
 };
 
+struct usnic_ib_create_qp_cmd {
+ struct usnic_transport_spec spec;
+};
+
 /*TODO: Future - usnic_modify_qp needs to pass in generic filters */
 struct usnic_ib_create_qp_resp {
  u32    vfid;
-- 
1.7.1